### PR TITLE
[FLINK-6173] [table] Clean-up flink-table jar and dependencies

### PR DESCRIPTION
**Pull request by twalthr**
## What is the purpose of the change

This PR cleans the `flink-table` dependencies almost entirely. The jar only contains files that are important. It shades everything except for Janino, because this causes problems. I think this PR makes the dependency management for users easier. The resulting jar looks like:

screen shot 2017-10-16 at 3 24 24 pm

I ran some smaller test programs and could not spot any problems, feel free to test it as well.

## Brief change log

Modification of the `flink-table` `pom.xml`.

## Verifying this change

Run all tests and run an example program on the cluster.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): yes
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no)
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

File: flink-libraries/flink-table/pom.xml [hunk 1]
```diff
@@ -252,6 +261,29 @@ under the License.
...
 								</relocation>
+								<relocation>
+									<pattern>org.pentaho</pattern>
+									<shadedPattern>org.apache.flink.shaded.calcite.org.pentaho</shadedPattern>
+								</relocation>
+
+								<!-- flink-table dependencies -->
+								<relocation>
+									<pattern>org.apache.commons</pattern>
+									<shadedPattern>org.apache.flink.shaded.org.apache.commons</shadedPattern>
```
Comment (zentol) [inline flink-libraries/flink-table/pom.xml]: For flink dependencies, please change the shading pattern to `org.apache.flink.table.shaded` to a) make them more unique and b) distinguish them from flink-shaded dependencies.

File: flink-libraries/flink-table/pom.xml [hunk 2]
```diff
@@ -252,6 +261,29 @@ under the License.
...
+								</relocation>
+								<relocation>
+									<pattern>org.reflections</pattern>
+									<shadedPattern>org.apache.flink.shaded.org.reflections</shadedPattern>
+								</relocation>
+								<relocation>
+									<pattern>org.joda.time</pattern>
+									<shadedPattern>org.apache.flink.shaded.org.joda.time</shadedPattern>
+								</relocation>
+								<!-- not relocated for now -->
```
Comment (zentol) [inline flink-libraries/flink-table/pom.xml]: why not?

File: flink-libraries/flink-table/pom.xml [hunk 3]
```diff
@@ -213,36 +209,49 @@ under the License.
...
-								<!--<relocation>
-									<pattern>org.slf4j</pattern>
-									<shadedPattern>org.apache.flink.shaded.calcite.org.slf4j</shadedPattern>
-								</relocation>-->
+								<!-- Calcite and its dependencies -->
 								<relocation>
-									<pattern>com.fasterxml.jackson</pattern>
-									<shadedPattern>org.apache.flink.shaded.calcite.com.fasterxml.jackson</shadedPattern>
+									<pattern>org.apache.calcite</pattern>
+									<shadedPattern>org.apache.flink.shaded.calcite.org.apache.calcite</shadedPattern>
```
Comment (zentol) [inline flink-libraries/flink-table/pom.xml]: For calcite dependencies, please change the shading pattern to org.apache.flink.calcite.shaded to a) make them more unique and b) distinguish them from flink-shaded dependencies.

File: flink-libraries/flink-table/pom.xml [hunk 4]
```diff
@@ -213,36 +209,49 @@ under the License.
...
 										<exclude>codegen/**</exclude>
 										<exclude>google/**</exclude>
 										<exclude>META-INF/*.SF</exclude>
 										<exclude>META-INF/*.DSA</exclude>
 										<exclude>META-INF/*.RSA</exclude>
+										<exclude>META-INF/services/**</exclude>
+										<exclude>properties.dtd</exclude>
+										<exclude>PropertyList-1.0.dtd</exclude>
+										<exclude>digesterRules.xml</exclude>
+										<!-- not excluded for now -->
```
Comment (zentol) [inline flink-libraries/flink-table/pom.xml]: why not?

File: flink-libraries/flink-table/pom.xml (hunk 2 above)
Comment (twalthr) [reply flink-libraries/flink-table/pom.xml]: The problem is that Calcite is using the Janino properties file for determining the location of the compiler. We could relocate Janino as well but then we also have to change the contents of the properties file. Have you done this before?

Comment (zentol) [reply flink-libraries/flink-table/pom.xml]: no, but that's exactly what you should write as a comment into the pom ;)

**Comment by zentol** [comment]
the table examples no longer build on travis.

**Comment by fhueske** [comment]
I'll fix the broken build and will merge this PR.

**Pull request closed without merging**